src/server/api/login/handlers.ts

```diff
@@ -73,7 +73,7 @@ export async function verifyOtp(req: Express.Request, res: Express.Response) {
   const { getOtpForEmail, setOtpForEmail, deleteOtpByEmail } = container.get<
     OtpCache
   >(DependencyIds.otpCache)
-  const { findOrCreateWithEmail } = container.get<UserRepository>(
+  const { findOrCreateWithEmail } = container.get<UserRepositoryInterface>(
```

Comment:
can we move all `container.get` calls to the top-level imports?

Reply:
They are currently in the function to allow test suites to inject dependencies before every function call. If it is injected outside the function, then they are bound during module import and can't be changed.

It can be solved once we refactor controllers/services to be objects and have them injected on instantiation.

Reply:
sounds good